### PR TITLE
Refactor tests for handshake.

### PR DESCRIPTION
**Pull request by drvinceknight**
For #884

### CODE REVIEW

File: axelrod/tests/strategies/test_handshake.py [hunk 1]
```diff
@@ -21,13 +21,21 @@ class TestHandshake(TestPlayer):
...
-        self.responses_test([D] * 20, [C, D], [C, C])
-        self.responses_test([D] * 20, [C, D], [D, C])
-        self.responses_test([D] * 20, [C, D], [D, D])
+        opponent = axelrod.MockPlayer([C, D])
+        actions = [(C, C), (D, D)] + [(C, C), (C, D)] * 10
+        self.versus_test(opponent, expected_actions=actions)
 
-        self.responses_test([D], [C, D] * 2, [D, C] * 2)
-        self.responses_test([C], [C, D] * 2, [C, D] * 2)
+        opponent = axelrod.MockPlayer([C])
```
Comment (marcharper) [inline axelrod/tests/strategies/test_handshake.py]: Cooperator?

File: axelrod/tests/strategies/test_handshake.py [hunk 2]
```diff
@@ -21,13 +21,21 @@ class TestHandshake(TestPlayer):
...
-        # Test initial play sequence
-        self.responses_test([C, D])
+        # Test first play
+        self.first_play_test(C)
 
-        self.responses_test([C] * 20, [C, D], [C, D])
-        self.responses_test([D] * 20, [C, D], [C, C])
-        self.responses_test([D] * 20, [C, D], [D, C])
-        self.responses_test([D] * 20, [C, D], [D, D])
+        opponent = axelrod.MockPlayer([C, D])
```
Comment (marcharper) [inline axelrod/tests/strategies/test_handshake.py]: alternator?

File: axelrod/tests/strategies/test_handshake.py [hunk 3]
```diff
@@ -21,13 +21,21 @@ class TestHandshake(TestPlayer):
...
-        self.responses_test([C], [C, D] * 2, [C, D] * 2)
+        opponent = axelrod.MockPlayer([C])
+        actions = [(C, C), (D, C)] + [(D, C)] * 20
+        self.versus_test(opponent, expected_actions=actions)
+
+        opponent = axelrod.MockPlayer([D, C])
+        actions = [(C, D), (D, C)] + [(D, D), (D, C)] * 10
+        self.versus_test(opponent, expected_actions=actions)
+
+        opponent = axelrod.MockPlayer([D])
```
Comment (marcharper) [inline axelrod/tests/strategies/test_handshake.py]: defector?

Comment (drvinceknight) [inline axelrod/tests/strategies/test_handshake.py]: Good call. No idea why I did this that way.

**Comment by drvinceknight** [comment]
Fixed.

**Pull request merged**